src/components/instance-form-builder/GeneralInfoForm.jsx

```diff
@@ -166,12 +166,21 @@ class GeneralInfoForm extends React.Component {
                 return;
             }
 
-            this.setState({ isSaving: true });
+            this.props.instance.validateInstanceName(this.props.d2).then(isValid => {
+                if (!isValid.status) {
+                    this.props.snackbar.error(
+                        i18n.t("This URL and Username combination already exists")
+                    );
+                    return;
+                }
 
-            this.props.originalInstance.remove(this.props.d2).then(() => {
-                this.props.instance.save(this.props.d2, this.props.isEdit).then(() => {
-                    this.setState({ isSaving: false });
-                    this.props.history.push("/instance-configurator");
+                this.setState({ isSaving: true });
+
+                this.props.originalInstance.remove(this.props.d2).then(() => {
+                    this.props.instance.save(this.props.d2, this.props.isEdit).then(() => {
+                        this.setState({ isSaving: false });
+                        this.props.history.push("/instance-configurator");
```

Comment:
Mental note, also for me, I didn't follow it in vaccination-app: those more-or-less generic components should not have hardcoded urls, it makes more sense to accept a prop `onSave` and let the parent decide what to do. No need to change it here.

Reply:
I can remove all of these references but I'll prefer to do that on a separate PR in order to not mix to much stuff here